### ISSUE TITLE
curve25519 used for end-to-end, but not for TLS

### ISSUE THREAD

**Issue by nisbet-hubbard** [element-hq/element-x-ios#3655]
**Is your feature request related to a problem? Please describe.**
I get it that curve25519 is being used for end-to-end encryption, which is good until https://github.com/matrix-org/vodozemac/pull/12 happens. But it’s not being used for TLS.

**Describe the solution you'd like**
So currently, this is what nginx says when it talks to the client:

```
prime256v1:secp384r1:secp521r1->prime256v1
```

If curve25519 was implemented for TLS, X25519 would appear instead.

**Describe alternatives you've considered**
Even when P256 is used for TLS, the end-to-end algorithm remains intact. So the alternative/status quo is to only use curve25519 in the end-to-end logic.

**Additional context**
https://spec.matrix.org/unstable/client-server-api/#end-to-end-encryption

**Comment by Laitinlok** [comment]
Yeah my nginx configuration only uses X25519 and the server is unreachable, however it does work with the Android client
